pkg/istio-agent/agent.go

```diff
@@ -435,12 +435,14 @@ func (a *Agent) FindRootCAForXDS() (string, error) {
 	} else if a.cfg.XDSRootCerts != "" {
 		// Using specific platform certs or custom roots
 		rootCAPath = a.cfg.XDSRootCerts
-	} else if fileExists("./etc/certs/root-cert.pem") {
+	} else if fileExists(security.DefaultRootCertFilePath) {
 		// Old style - mounted cert. This is used for XDS auth only,
 		// not connecting to CA_ADDR because this mode uses external
 		// agent (Secret refresh, etc)
-		return "./etc/certs/root-cert.pem", nil
-	} else if a.secOpts.PilotCertProvider == "kubernetes" {
+		return security.DefaultRootCertFilePath, nil
+	} else if a.secOpts.PilotCertProvider == constants.CertProviderNone {
```

Comment:
should this be last? Seems like if you use file mounted certs or prov certs its fine to have this be "none"

Reply:
Thanks! Make sense to me. Changed to the last of the specified in terms of ordering.